openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -122,25 +122,25 @@ public void prepare(TestCase test) {
     validateTestParameters(test.getParameterValues(), testDefinition.getParameterDefinition());
   }
 
-  private EntityReference getTestSuite(TestCase test) {
+  private EntityReference getTestSuite(TestCase test) throws EntityNotFoundException {
     // `testSuite` field returns the executable `testSuite` linked to that testCase
     List<CollectionDAO.EntityRelationshipRecord> records =
         findFromRecords(test.getId(), entityType, Relationship.CONTAINS, TEST_SUITE);
-    ensureSingleRelationship(entityType, test.getId(), records, Relationship.CONTAINS.value(), true);
     for (CollectionDAO.EntityRelationshipRecord testSuiteId : records) {
       TestSuite testSuite = Entity.getEntity(TEST_SUITE, testSuiteId.getId(), "", Include.ALL);
       if (Boolean.TRUE.equals(testSuite.getExecutable())) {
         return testSuite.getEntityReference();
       }
     }
-    return null;
+    throw new EntityNotFoundException(
+        String.format("Error occurred when retrieving executable test suite for testCase %s. ", test.getName())
+            + "No executable test suite was found.");
   }
 
   private List<TestSuite> getTestSuites(TestCase test) {
     // `testSuites` field returns all the `testSuite` (executable and logical) linked to that testCase
     List<CollectionDAO.EntityRelationshipRecord> records =
         findFromRecords(test.getId(), entityType, Relationship.CONTAINS, TEST_SUITE);
-    ensureSingleRelationship(entityType, test.getId(), records, Relationship.CONTAINS.value(), true);
```

Comment:
we should ensureSingleRelationship, for testcase -> testSuite that is executable no?

Reply:
Wouldn't we fall in the same case though as we'll list all all the relationships to the testSuites in the method?